lib/repo/git_repository.rb

```diff
@@ -1077,9 +1077,13 @@ def path_exists?(path)
       true
     end
 
-    # Return changed files at 'path' (recursively)
+    # Return changed files at 'path'
     def changed_files_at_path(path)
-      return files_at_path_helper(path, true)
+      files = files_at_path(path)
```

Comment:
files_at_path doesn't include directories.  So I think you need to explicitly check the directories_at_path too.